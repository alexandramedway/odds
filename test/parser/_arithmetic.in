state 1 + 2.1
state 1 - 2.1
state 1 * 2.1
state 1 / 2.1
state 1 % 2.1
state 1 ** 2.1
state -42
state 1 + -43
state 1 * 2 + 3 ** 4
state 1 / 2 % 3 % 4
state 1 + 2 - 3 / 4
state 1 * (2 + 3)
