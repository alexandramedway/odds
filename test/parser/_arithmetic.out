[
	Do(Binop(Int_lit(1), Add, Float_lit(2.1))) ;
	Do(Binop(Int_lit(1), Sub, Float_lit(2.1))) ;
	Do(Binop(Int_lit(1), Mult, Float_lit(2.1))) ;
	Do(Binop(Int_lit(1), Div, Float_lit(2.1))) ;
	Do(Binop(Int_lit(1), Mod, Float_lit(2.1))) ;
	Do(Binop(Int_lit(1), Pow, Float_lit(2.1))) ;
	Do(Unop(Sub, Int_lit(42))) ;
	Do(Binop(Int_lit(1), Add, Unop(Sub, Int_lit(43)))) ;
	Do(
		Binop(
			Binop(Int_lit(1), Mult, Int_lit(2)),
			Add,
			Binop(Int_lit(3), Pow, Int_lit(4))
		)
	) ;
	Do(
		Binop(
			Binop(
				Binop(Int_lit(1), Div, Int_lit(2)),
				Mod,
				Int_lit(3)
			),
			Mod,
			Int_lit(4)
		)
	) ;
	Do(
		Binop(
			Binop(Int_lit(1), Add, Int_lit(2)),
			Sub,
			Binop(Int_lit(3), Div, Int_lit(4))
		)
	) ;
	Do(
		Binop(
			Int_lit(1),
			Mult,
			Binop(Int_lit(2), Add, Int_lit(3))
		)
	)
]
